src/binance_websocket.cpp

```diff
@@ -142,15 +142,17 @@ void binance::Websocket::connect_endpoint(CB cb, const char* path)
 }
 
 // Entering event loop
-void binance::Websocket::enter_event_loop()
+void binance::Websocket::enter_event_loop(std::chrono::hours hours)
 {
-	while (1)
-	{	
-		lws_service(context, 500);
-
-		if (lws_service_cancelled)
-			break;
-	}
+    auto start = std::chrono::high_resolution_clock::now();
+    auto end = start + hours;
+    do {
+        using namespace std::chrono;
+
+        lws_service(context, 500);
+        if (lws_service_cancelled)
+            break;
+    } while (std::chrono::high_resolution_clock::now() < end);
```

Comment:
High-resolution clock is maybe too precise for this situation? Low-resolution clock would put less workload on the system

Reply:
Yes I agree on using Low-resolution clock, the system will benefit from it.